### PR TITLE
Tests rework and code clean up

### PR DESCRIPTION
**Pull request by darkgl0w**
Hello,

This PR essentially adds some tests to improve the code coverage and to ensure that the plugin acts as it should.

It also splits the tests in 3 different files in order to group them by scope making it easier to read, to understand and to avoid ending with a big test file ^^.

Feel free to ping me if changes need to be made ^^

PS: dependencies have been updated too.

#### Checklist

- [x] run `npm run test`
- [x] tests are included
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
 function transactionUtil (pool, fn, cb) {
   pool.connect((err, client, done) => {
     if (err) return cb(err)
 
     const shouldAbort = (err) => {
       if (err) {
-        client.query('ROLLBACK', () => {
-          done()
-        })
+        client.query('ROLLBACK', () => done())
```
Comment (mcollina) [inline index.js]:
```suggestion
        client.query('ROLLBACK', done)
```

File: index.js [hunk 2]
```diff
@@ -63,7 +60,7 @@ function fastifyPostgres (fastify, options, next) {
   if (options.native) {
     delete options.native
     if (!pg.native) {
-      console.warn('pg-native not installed, can\'t use native option - fallback to pg module')
+      console.warn(`pg-native not installed, can't use native option - fallback to pg module`)
```
Comment (mcollina) [inline index.js]: I prefer to use interpolation _only_ when there is some interpolation going on.

File: test/query.test.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const Fastify = require('fastify')
+const fastifyPostgres = require('../index')
+
+test('When fastify.pg root namespace is used:', (childTest) => {
```
Comment (mcollina) [inline test/query.test.js]: I would just call this variable `t`.

File: index.js (hunk 2 above)
Comment (darkgl0w) [reply index.js]: Applied double quotes instead, like this there is no need to escape the single quote ^^

File: test/query.test.js (hunk 3 above)
Comment (darkgl0w) [reply test/query.test.js]: done ^^

File: index.js (hunk 2 above)
Comment (mcollina) [reply index.js]: we use single quotes, so let's keep that escaped.

Comment (darkgl0w) [reply index.js]: done ^^

**Comment by mcollina** [comment]
Good work, thanks!

**Pull request merged**